internal/dot/graph.go

```diff
@@ -25,15 +25,26 @@ import (
 	"reflect"
 )
 
+// State of a constructor or group is updated when they fail to build.
+type State int
+
+const (
+	initial State = iota
+	pointOfFailure
+	transitiveFailure
```

Comment:
nit: FailureState? initial can be `success`, pointOfFailure can be `rootCause` and transitiveFailure can be `failed`.

Reply:
Renamed the failure states to `success`, `rootCause`, and `failed` everywhere.